Drop D1 support and use swarm v6.0.0 RC

tiyash-basu-sociomantic opened the pull request.

tiyash-basu-sociomantic committed:
Apply d1to2fix (everything except `this` -> `(&this`))

Note: The refactors in this commit introduce compilation errors,
which should be resolved in a few successive commits

tiyash-basu-sociomantic committed:
Apply d1to2fix (only `this` -> `(&this`))

Note: The refactors in this commit introduce compilation errors,
which should be resolved in a few successive commits.

The changes introduced by d1to2fix in this case have been mostly
reverted, wherever `(&this)` (changed from `this`) added an unnecessary
indirection.

The only case remaining in this commit is the one where the indirection
is specifically needed.

tiyash-basu-sociomantic committed:
Advance `beaver` to `v0.6.5`

* submodules/beaver v0.5.2(e3b3351)...v0.6.5(c2c7a05) (14 commits)
  > dlang/d2-release: Fix wrong option for git add
  > dlang/d2-release: Mark repository as D2_ONLY
  > Fix typos in README
  > Fix nested test condition
  > Update copyright owner
  (...)

# Old submodules summary:
#
# * submodules/beaver v0.5.2(e3b3351)...v0.6.5(c2c7a05) (14 commits)
#   > dlang/d2-release: Fix wrong option for git add
#   > dlang/d2-release: Mark repository as D2_ONLY
#   > Fix typos in README
#   > Fix nested test condition
#   > Update copyright owner
#   (...)

tiyash-basu-sociomantic committed:
Advance `ocean` to `v5.0.4`

* submodules/ocean v4.0.11(b85bb255)...v5.0.4(5e47d4df) (150 commits)
  > Merge tag v4.6.2 into v5.0.x
  > Explicitly handle `void*` types in `ReduceType`
  > Add sanity tests of instantiations of `ReduceType`
  > Merge tag v4.6.1 into v5.0.x
  > Test with upstream DMD 2.085
  (...)

tiyash-basu-sociomantic committed:
Advance `turtle` to `v10.0.3`

* submodules/turtle v9.0.2(c486df9)...v10.0.3(c10a254) (26 commits)
  > Merge tag v9.1.5 into v10.0.x
  > Switch to more recent upstream DMD in CI
  > Upgrade ocean to get rid of deprecations
  > Release note
  > Adapt CI to be D2 only
  (...)

tiyash-basu-sociomantic committed:
Drop D1 support from build directives

including in the CI.

tiyash-basu-sociomantic committed:
Make request classes implement the `IRequest` interface

...instead of the `IRequestHandler` interface, which has now been removed
from swarm.

tiyash-basu-sociomantic committed:
Rename `IRequestHandlerRequest` to 'RequestHandler`

tiyash-basu-sociomantic committed:
Implement `handler` method for neo request classes

This commit collates the functionalities of `preSupportedCodeSent` and
`preSupportedCodeSent` methods in the implementation of `IRequest.handle`
method.

tiyash-basu-sociomantic committed:
Stop fakedmq node after test

tiyash-basu-sociomantic committed:
Add tests with dmd v2.086.* to CI

matthias-wende-sociomantic commented on .travis.yml:
```diff
@@ -39,6 +39,10 @@ jobs:
         - <<: *test-matrix
           env: DMD=2.078.3.s* F=devel
+        - <<: *test-matrix
+          env: DMD=2.085.* F=production
```
Is there a reason for using 2.085 in particular and not 2.086?

tiyash-basu-sociomantic replied:
There wasn't a specific reason. I've changed it to 2.086.*.

matthias-wende-sociomantic commented on src/fakedmq/Storage.d:
```diff
@@ -115,7 +115,7 @@ public struct Storage
 
         Params:
             name = channel name
-    
```
Just a reminder. This seems unrelated.

tiyash-basu-sociomantic committed:
Advance `swarm` to `v6.0.0-rc.3`

* submodules/swarm v5.0.20(186de47)...v6.0.0-rc.3(fa7475d) (118 commits)
  > Add static checks for members of `Request` type in `ConnectionHandler`
  > Remove deprecated static ifs
  > Explicitly exclude pointers in `StructPacker`
  > Remove unused module `swarm.neo.util.VoidBufferAsArrayOf`
  > Use `ocean.util.container.VoidBufferAsArrayOf`
  (...)

tiyash-basu-sociomantic commented:
Updated to use swarm `v6.0.0-rc.3`.

matthias-wende-sociomantic commented on src/dmqproto/node/neo/request/Pop.d:
```diff
@@ -3,7 +3,7 @@
 
     Copyright:
-        Copyright (c) 2016-2017 dunnhumby Germany GmbH. All rights reserved.
+        Copyright (c) 2016-2017 dunnhumby Germany GmbH. All rights reservthis.ed.
```
reservthis.ed.

tiyash-basu-sociomantic replied:
:man_facepalming:

matthias-wende-sociomantic commented on src/dmqproto/node/neo/request/Push.d:
```diff
@@ -3,7 +3,7 @@
 
     Copyright:
-        Copyright (c) 2016-2017 dunnhumby Germany GmbH. All rights reserved.
+        Copyright (c) 2016-2017 dunnhumby Germany GmbH. All rights reservthis.ed.
```
ditto

matthias-wende-sociomantic reviewed (approved):
Two tiny comments :).
